### PR TITLE
add interactive colorbar example to gallery

### PR DESCRIPTION
**Pull request by richardsheridan**
## PR Summary
Added an example to the gallery for discussion regarding #19515.  All that needs to happen in each event is to modify the colorbar norm vmin and vmax, although the particular interactivity that was desired in #19515 would require a little more creativity in the callbacks.

Hoping for feedback as to: 

1. whether to actually include in gallery
2. style of new gallery page
3. code review

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: examples/userdemo/colormap_interactive_adjustment.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+"""
+========================================
+Interactive Adjustment of Colormap Range
+========================================
+
+Demonstration of using colorbar and picker functionality to make an
+interactively adjustable colorbar widget.
+"""
```
Comment (timhoffm) [inline examples/userdemo/colormap_interactive_adjustment.py:8]: Please add a paragraph describing how this is used.

File: examples/userdemo/colormap_interactive_adjustment.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.backend_bases import MouseButton
+
+
+def pick_fn(event):
```
Comment (timhoffm) [inline examples/userdemo/colormap_interactive_adjustment.py:15]:
```suggestion
def on_pick(event):
```

is sort of a convention for event handling functions.

File: examples/userdemo/colormap_interactive_adjustment.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    adjust_colorbar(event.mouseevent)
+
+
+def motion_fn(mouseevent):
```
Comment (timhoffm) [inline examples/userdemo/colormap_interactive_adjustment.py:19]:
```suggestion
def on_move(mouseevent):
```

File: examples/userdemo/colormap_interactive_adjustment.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        # discard all others
+        return
+
+    canvas.draw_idle()
+
+
+fig, ax = plt.subplots()
+canvas = fig.canvas
+arr = np.random.random((100, 100))
+axesimage = plt.imshow(arr)
```
Comment (timhoffm) [inline examples/userdemo/colormap_interactive_adjustment.py:39]: AFAICS, the data range mapping to the full color scale is reduced by the interactive selection. As a result, we'll have over/under data. Unfortunately the default colormap marks these with the regular max/min colorbar values. This is quite confusing in the scaling context. I propose to mark out-of scale values:

```suggestion
cmap = plt.get_cmap('viridis').with_extremes(under='darkgrey', over='grey')
axesimage = plt.imshow(arr, cmap=cmap)
```

**Comment by timhoffm** [comment]
BTW thanks for taking the time to put this example together!

**Comment by richardsheridan** [comment]
I feel #20479 blocks this PR so marking as draft until resolved.

**Comment by richardsheridan** [comment]
Also something about the new `ColorbarAxes` detaches the mouse event xdata and ydata from the data range, so this only coincidentally worked since the range of the data was 0..1, another blocker.

**Comment by richardsheridan** [comment]
Unblocked by #20501, thanks for everyone's help there! (Somehow I didn't get any notifications about that PR?)

**Comment by timhoffm** [comment]
flake8:
> ./examples/userdemo/colormap_interactive_adjustment.py:51:80: E501 line too long (87 > 79 characters)

**Comment by jklymak** [comment]
Maybe I'm doing something wrong, but this only works for one click for me and then the colorbar stops responding.  Is that expected, or am I doing something wrong?

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Thanks!

**Pull request merged**